### ISSUE TITLE
[Task]: Use python uv to run lint/test/check commands

### ISSUE THREAD

**Issue by KevinMind** [mozilla/addons#15210]
### Description

UV is a new python package manager/runner that (one could argue) is an improvement to pip.

Instead of migrating everything all at once, we could try to use uv to run pytest and ruff. The nice thing about UV is you can run pythyon executables without pre-installing the dependencies, UV will install them for you. That is great for these two deps in particular because it means you could run pytest/ruff from a host machine instead of needing a running environment.

That's kinda cool. This would also give us some testing room to try out the system and if we like it and it makes sense, we could follow up by replacing pip entirely.

One downside could be that running pytest outside of a container could change the expected outcome of tests, if the tests themselves depend on specific environment properties. That could be a feature or a bug, depending on your viewpoint, but would be good to know which tests are container dependent in any case.

### Acceptance Criteria

```[tasklist]
  ### Milestones/checkpoints
  - [ ] uv installed via make
  - [ ] move ruff calls outside of Makefile-docker to -os and use uv to install/execute
  - [ ] move pytest calls outside of Makefile-docker to -os and use uv to install/execute
  - [ ] Update relevant CI logic that may not need to depend on a built image (like check)
```

### Checks

- [X] If I have identified that the work is specific to a repository, I have removed "repository:addons-server" or "repository:addons-frontend"

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-1274)

**Comment by eviljeff** [comment]
> One downside could be that running pytest outside of a container could change the expected outcome of tests, if the tests themselves depend on specific environment properties. That could be a feature or a bug, depending on your viewpoint, but would be good to know which tests are container dependent in any case.

Yeah... I don't think we should encourage running any addons-server code outside of the docker container.

**Comment by KevinMind** [comment]
We should split this ticket to focus on:
- installing dependencies
- executing commands

And perhaps focus first on replacing pip for installing dependencies.

**Comment by diox** [comment]
> And perhaps focus first on replacing pip for installing dependencies.

Something to be careful about if/when we do this: by default, `pip` compiles python code in bytecode, and [`uv` doesn't](https://pythonspeed.com/articles/faster-pip-installs/#with-matching-settings-uvs-performance-lead-declines), so it's faster to install deps but then the app might be slower to start. We'd probably want to flip the option to enable compilation at install time with `uv` too, which means there might not be as much speed gains (it might even be slower).
